helm/cluster-openstack/templates/_helpers.tpl

```diff
@@ -72,3 +72,38 @@ room for such suffix.
 {{- define "kubeProxyPreKubeadmCommands" -}}
 - bash /run/kubeadm/gs-kube-proxy-patch.sh
 {{- end -}}
+
+{{/*
+OpenStackMachineTemplate is immutable. We need to create new versions during upgrades.
+Here we are generating a hash suffix to trigger upgrade when only it is necessary by
+using only the parameters used in _openstack_machine_template.tpl.
+*/}}
+{{- define "osmtRevision" -}}
+{{- $inputs := (dict
```

Comment:
Does this selection of values make sense? As far as I can tell this dict defines in which case we try to do template rotation. How would this work if we e.g. upgrade the api version from v1beta1 to v1beta2?

Reply:
`Does this selection of values make sense? ` Those are the ones used in `OpenStackMachineTemplate`. When you change one of those, the template will be different. We have to change the name since OpenStackMachineTemplate is immutable. 

I tested parameter updates but didn't test version updates. However, shouldn't it be straightforward? We will update the templates with the new version. Even if we don't change any parameter, objects will have a different API version so those will be created from scratch (no update request by helm).

Reply:
> When you change one of those, the template will be different. 
Okay so this is basically all inputs we allow for OpenStackMachineTemplate then. So if we add new variables there, we need to make sure to remember :D 

> Even if we don't change any parameter, objects will have a different API version so those will be created from scratch (no update request by helm).
I am not sure that's true because they will have the same name/namespace. I might be wrong here :)

Reply:
Okey. I was wrong. When we just bump the versions, it will update the existing CRs. When necessary, we can adapt the revision function (e.g. adding version param as well) to generate something different so that we can upgrade withot collision.

Reply:
> we can adapt the revision function (e.g. adding version param as well)

But then we'll recreated machinedeployments on every version bump regardless of whether that makes sense or no, right?

But for apiversion change we can add artificial `"apiVersion": "v1beta2"` and that will change the hash.